libcaf_core/caf/detail/json.cpp

```diff
@@ -67,8 +110,35 @@ size_t do_unescape(const char* i, const char* e, char* out) {
             case 'v':
               *out++ = '\v';
               break;
+            case 'u': {
+              // The parser will enforce 4 hex chars after an '\u' escape.
+              auto code_point = uint32_t{0};
+              read_ascii(code_point, i);
+              // Test if we decoded the first value of an UTF-16 surrogate pair.
+              if (code_point >= 0xD800 && code_point < 0xDC00) {
+                // A low surrogate must follow. This isn't always enforced by
+                // the parser we have to check at this point.
+                if (std::distance(i, e) > 6 && i[1] == '\\' && i[2] == 'u') {
+                  i += 2;
```

Comment:
The iterator is in fact pointing to the last character of the last sequence (before `\`). This is done so that the `++i` below the switch statement can be executed properly (even if there is no second surrogate).
The `read_ascii` function uses `++c` instead of `c++` so that `i+=2` is in fact correct. I see the confusion here, I'll refactor.